src/libs/ReportUtils.ts

```diff
@@ -3464,14 +3477,63 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+function transactionHasViolation(transactionID: string, transactionViolations: TransactionViolations): boolean {
+    const violations: TransactionViolation[] = transactionViolations[transactionID];
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ * Checks to see if a report's parentAction is a money request that contains a violation
+ */
+function transactionThreadHasViolations(report: Report, transactionViolations: TransactionViolations, reportActions: OnyxCollection<ReportActions> | null = {}): boolean {
```

Comment:
Same here `hasTransactionThreadViolations`

Reply:
I don't think this is a good name, because there's no such thing as a "transaction-thread violation", so that's a little confusing. 

If we're going to change it, I suggest `ReportUtils.doesTransactionThreadHaveViolations`

Reply:
It's now `ReportUtils.doesTransactionThreadHaveViolations`